### ISSUE TITLE
Opening the second dialog when the first dialog opens, the UI gets unresponsible.

### ISSUE THREAD

**Issue by tohkawaics** [vaadin/vaadin-dialog-flow#121]
Opening the second dialog when the first dialog opens, the UI gets unresponsible.

MainView.java:
```
@Route
public class MainView extends VerticalLayout {
  public MainView() {
    Button button = new Button("open", buttonClickEvent -> {
      DialogA dialogA = new DialogA();
      dialogA.open();
    });
    add(button);
  }
}
```

DialogA.java:
```
public class DialogA extends Dialog {
  public DialogA() {
    H1 h1 = new H1();
    h1.setText("DialogA");
    add(h1);
    Button button = new Button("close", buttonClickEvent -> {
      close();
    });
    setCloseOnEsc(false);
    setCloseOnOutsideClick(false);
    add(button);
    DialogB dialogB = new DialogB();
    dialogB.open();
  }
}
```

DialogB.java:
```
public class DialogB extends Dialog {
  public DialogB() {
    H1 h1 = new H1();
    h1.setText("DialogB");
    add(h1);
    Button button = new Button("open", buttonClickEvent -> {
      close();
    });
    add(button);
  }
}
```

The problem seems to be introduced from Vaadin 12.0.6 (and 13.0.0).

~~Until 12.0.5, it worked as expected.~~  I suspect this is the regression of #113.

**Comment by denis-anisimov** [comment]
Reproduced with `12.0.6` , `12.0.7` Platform  (using skeleton starter).

The UI behavior is:
- click `open` button. The first dialog is opened and you may not able to do anything anymore. Clicks doesn't close the dialog.

I'm not sure about the expected behavior.

I use this test in the dialog repository (`master` branch ):

```
@Route("two-dialogs")
public class TwoDialogsPage extends Div {

    public TwoDialogsPage() {
        NativeButton button = new NativeButton("open", buttonClickEvent -> {
            FirstDialog dialog = new FirstDialog();
            dialog.open();
        });
        add(button);
    }

    public static class FirstDialog extends Dialog {
        public FirstDialog() {
            H1 h1 = new H1();
            h1.setText("DialogA");
            add(h1);
            NativeButton button = new NativeButton("close",
                    buttonClickEvent -> close());
            setCloseOnEsc(false);
            setCloseOnOutsideClick(false);
            add(button);
            SecondDialog dialog = new SecondDialog();
            dialog.open();
        }
    }

    public static class SecondDialog extends Dialog {
        public SecondDialog() {
            H1 h1 = new H1();
            h1.setText("DialogB");
            add(h1);
            NativeButton button = new NativeButton("close",
                    buttonClickEvent -> close());
            add(button);
        }
    }
}
```

The result is: the first dialog is opened, the second one is not until I close the first dialog.
The first dialog may be closed so UI is at least responsive.

I'm pretty sure that the issue is on the client side.
Also I may suggest that it works in master because it uses the latest client side dialog component version where the behavior is fixed.
Perhaps `12.0.6` uses the dialog version with this bug.

**Comment by tohkawaics** [comment]
Sorry, my comment 'Until 12.0.5, it worked as expected' was incorrect.

The behavior on 12.0.5 was the same as yours on master branch.

My expected behavior is

1. The first dialog opens (and fills the form with the initial values) 
1. The second dialog is automatically opens over the first dialog (for example, when there was some kind of warning messages while filling the form)
1. When close the second dialog, the first dialog gets active

**Comment by tohkawaics** [comment]
I should use attachListener to open the second dialog automatically.

This is the code to represent this bug (sorry, my first example was messed up...)
```
@Route("two-dialogs")
public class TwoDialogsPage extends Div {

  public TwoDialogsPage() {
    NativeButton button = new NativeButton("open", buttonClickEvent -> {
      FirstDialog dialog = new FirstDialog();
      dialog.open();
    });
    add(button);
  }

  public static class FirstDialog extends Dialog {
    public FirstDialog() {
      H1 h1 = new H1();
      h1.setText("DialogA");
      add(h1);
      NativeButton button = new NativeButton("close",
              buttonClickEvent -> close());
      setCloseOnEsc(false);
      setCloseOnOutsideClick(false);
      add(button);
      addAttachListener(attachEvent -> {
        SecondDialog dialogB = new SecondDialog();
        dialogB.open();
      });
    }
  }

  public static class SecondDialog extends Dialog {
    public SecondDialog() {
      H1 h1 = new H1();
      h1.setText("DialogB");
      add(h1);
      NativeButton button = new NativeButton("close",
              buttonClickEvent -> close());
      add(button);
      setCloseOnEsc(false);
      setCloseOnOutsideClick(false);
    }
  }
}
```

**Comment by brp-benjamin** [comment]
We are having issues with this bug. Following is an investigation protocol using the supplied minimal reproducible example:

 * Vaadin 12.0.5 works as expected
 * Vaadin 12.0.6 does not.
 * The bug is also present in vaadin 13.0.4.

When inspecting the bom the following dependencies were changed between 12.0.5 and 12.0.6:

com.vaadin » vaadin-spring-bom
com.vaadin » vaadin
com.vaadin » vaadin-core
com.vaadin » vaadin-testbench
com.vaadin » vaadin-spring-boot-starter
com.vaadin » vaadin-checkbox-flow
com.vaadin » vaadin-combo-box-flow
com.vaadin » vaadin-radio-button-flow
com.vaadin » vaadin-upload-flow
org.webjars.bowergithub.vaadin » vaadin-combo-box
org.webjars.bowergithub.vaadin » vaadin-overlay

Vaadin-overlay seems like the likely culprit in this case.

Using vaadin 13.0.4 with a downgraded vaadin-overlay from 12.0.5 (vaadin-overlay 3.2.5) removes the bug. We have locally resolved by downgrading to vaadin-overlay 3.2.5, presumably this introduces other bugs that were fixed in vaadin-overlay 3.2.11 so it is not an ideal solution.

I have created a corresponding bug report in vaadin-overlay:

https://github.com/vaadin/vaadin-overlay/issues/152
